# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/TimeUnits.cpp
# Commit: 0d9c1ac74d9c
# Full Hash: 0d9c1ac74d9cf147aa99fad91c7e1eca0e24fa1b
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-02 04:43:27
# Description:
#   Bug 1874723 - Accept imprecision when trimming audio content media in certain cases. r=alwu
#   
#   In the test case, the time base is 69 (generated by a fuzzer), and so there's a
#   very large loss in precision in the computation, and eventually the assumption
#   doesn't hold. It should however hold for the common case.
# ==============================================================================

diff -r f84c76f9460d -r 0d9c1ac74d9c dom/media/TimeUnits.cpp
--- a/dom/media/TimeUnits.cpp	Thu Feb 01 17:10:37 2024 +0000
+++ b/dom/media/TimeUnits.cpp	Thu Feb 01 17:13:15 2024 +0000
@@ -98,6 +98,8 @@
   return mTicks.value() * aRate / mBase;
 }
 
+bool TimeUnit::IsBase(int64_t aBase) const { return aBase == mBase; }
+
 double TimeUnit::ToSeconds() const {
   if (IsPosInf()) {
     return PositiveInfinity<double>();